bb-core/src/main/java/com/cognifide/qa/bb/loadable/hierarchy/WebElementInterceptor.java

```diff
@@ -57,29 +57,33 @@
   public Object invoke(MethodInvocation methodInvocation) throws Throwable {
     Class methodCallerClass = getMethodCallerClassWithoutGuiceContext();
 
-    if (methodCallerClass.isAnnotationPresent(PageObject.class)) {
+    if (isApplicable(methodCallerClass)) {
 
-      LOG.
-        debug("Caught invocation of method {} from {}. Started processing loadable component conditions hierarchy",
+      LOG.debug(
+          "Caught invocation of method {} from {}. Started processing loadable component conditions hierarchy",
           methodInvocation.getMethod().getName(), methodCallerClass.getName());
 
       BobcatWebElement caller = (BobcatWebElement) methodInvocation.getThis();
       ClassFieldContext directContext = acquireDirectContext(caller);
-      ConditionStack loadableContextHierarchy
-        = loadableCondsExplorer.
-        discoverLoadableContextHierarchy(directContext, pageObjectInvocationTracker.
-          getSubjectStack());
+      ConditionStack loadableContextHierarchy =
+          loadableCondsExplorer.discoverLoadableContextHierarchy(directContext,
+              pageObjectInvocationTracker.getSubjectStack());
 
       loadConditionChainRunner.chainCheck(loadableContextHierarchy);
     }
     return methodInvocation.proceed();
 
   }
 
+  private boolean isApplicable(Class methodCallerClass) {
+    return methodCallerClass.isAnnotationPresent(PageObject.class)
+        || LoadableComponentsUtil.isStepsImplementationClass(methodCallerClass)
```

Comment:
I don't like the idea of a core functionality being in any way dependent on the runner used in the tests. If we add support for an additional framework, we would potentially have to touch this class again and this indicates a design problem.
The above should be addressed in the module related to the runner itself (in this case, `Bobcumber`).